tests/components/aussie_broadband/test_config_flow.py

```diff
@@ -0,0 +1,322 @@
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    with patch("aussiebb.asyncio.AussieBB.__init__", return_value=None), patch(
+        "aussiebb.asyncio.AussieBB.login", return_value=True
+    ), patch("aussiebb.asyncio.AussieBB.get_services", return_value=FAKE_SERVICES):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            FAKE_DATA,
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["step_id"] == "service"
+    assert result2["errors"] is None
+
+    with patch(
+        "homeassistant.components.aussie_broadband.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_SERVICES: [FAKE_SERVICES[1]["service_id"]]},
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result3["title"] == TEST_USERNAME
+    assert result3["data"] == FAKE_DATA
+    assert result3["options"] == {
+        CONF_SERVICES: [FAKE_SERVICES[1]["service_id"]],
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test invalid auth is handled."""
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("aussiebb.asyncio.AussieBB.__init__", return_value=None), patch(
+        "aussiebb.asyncio.AussieBB.login", side_effect=AuthenticationException()
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            FAKE_DATA,
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_network_issue(hass: HomeAssistant) -> None:
+    """Test network issues are handled."""
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("aussiebb.asyncio.AussieBB.__init__", return_value=None), patch(
+        "aussiebb.asyncio.AussieBB.login", side_effect=ClientConnectionError()
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            FAKE_DATA,
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_reauth(hass: HomeAssistant) -> None:
+    """Test reauth flow."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    # Test reauth but the entry doesn't exist
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_REAUTH}, data=FAKE_DATA
```

Comment:
Please start the reauth flow with the complete context that is passed by the config entry reauth helper method when raising `ConfigEntryAuthFailed`.